dev/lib/product_taxonomy/models/mixins/indexed.rb

```diff
@@ -67,6 +69,20 @@ def find_by(**conditions)
       hashed_models[field][value]
     end
 
+    # Find a model by field value. Returns the first matching record or raises an error if not found. Only works for
+    # fields marked unique.
+    #
+    # @param conditions [Hash] Hash of field-value pairs to search by
+    # @return [Object] The matching model
+    def find_by!(**conditions)
+      record = find_by(**conditions)
```

Comment:
Should we use `...` here or do we want the conditions splat?

```rb
def find_by!(...)
  record = find_by(...)
```

Reply:
Yep exactly, I tried the `...` first and then realized we need the params